packages/@aws-cdk/aws-s3objectlambda/lib/object-lambda.ts

```diff
@@ -0,0 +1,166 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as s3 from '@aws-cdk/aws-s3';
+import { Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnAccessPoint } from './s3objectlambda.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+/**
+  * Creates an S3 Object Lambda, which can intercept and transform
+  * `GetObject` requests.
+  *
+  * @param fn The Lambda function
+  * @param props Configuration for this Object Lambda
+  */
+export interface ObjectLambdaProps {
+  /**
+   * The bucket to which this Object Lambda belongs
+   */
+  readonly bucket: s3.IBucket
+
+  /**
+   * The Lambda function used to transform objects.
+   */
+  readonly fn: lambda.IFunction
+
+  /**
+   * The name of the Object Lambda access point.
+   */
+  readonly name: string
+
+  /**
+   * Whether CloudWatch metrics are enabled for the Object Lambda.
+   *
+   * @default false
+   */
+  readonly cloudWatchMetricsEnabled?: boolean
+
+  /**
+   * Whether the Lambda function can process `GetObject-Range` requests.
+   *
+   * @default false
+   */
+  readonly supportsGetObjectRange?: boolean
+
+  /**
+   * Whether the Lambda function can process `GetObject-PartNumber` requests.
+   *
+   * @default false
+   */
+  readonly supportsGetObjectPartNumber?: boolean
+
+  /**
+   * Additional JSON that provides supplemental data passed to the
+   * Lambda function on every request.
+   *
+   * @default - No data.
+   */
+  readonly payload?: string
+}
+
+/**
+  * An S3 Object Lambda for intercepting and transforming `GetObject` requests.
+  */
+export class ObjectLambda extends CoreConstruct {
```

Comment:
@duarten thanks for the PR and sorry it took so long to review! The main comment I have is that I think this PR should introduce an `AccessPoint` construct instead of `ObjectLambda` since that is the name of the corresponding L1. Also instead of extending `CoreConstruct` this should extend `Resource`. I would recommend taking a look at the example [here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/example-construct-library/lib/example-resource.ts) which has the base setup we would be looking for.

Reply:
Yeah and they are in different libraries, so you would use them like this. 

```ts
import * as objectlambda from '@aws-cdk/aws-s3objectlambda';
import * as s3 from '@aws-cdk/aws-s3';

new objectlambda.AccessPoint();
new s3.AccessPoint();

```

In this  case `S3ObjectLambda` is the name of the library and `AccessPoint` is the name of the class.

Reply:
Ah, makes sense! I'll update the PR asap.

Reply:
Updated!